erts/doc/src/erlang.xml

```diff
@@ -52,6 +59,300 @@
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="list" n_vars="0"/>
+      <desc>
+        <p>An Erlang <seeguide marker="system/reference_manual:data_types#list">list</seeguide>
+          containing terms of any type.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="list" n_vars="1"/>
+      <desc>
+        <p>An Erlang <seeguide marker="system/reference_manual:data_types#list">list</seeguide>
+          containing terms of the type <c><anno>ContentType</anno></c>.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="map" n_vars="0"/>
+      <desc>
+        <p>An Erlang <seeguide marker="system/reference_manual:data_types#map">map</seeguide>
+          containing any number of key and value associations.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="maybe_improper_list" n_vars="0"/>
+      <desc>
+        <p>An Erlang <seeguide marker="system/reference_manual:data_types#list">list</seeguide>
+          containing terms any type that may not end with a <seetype marker="#nil">[]</seetype>.
+        </p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="maybe_improper_list" n_vars="2"/>
+      <desc>
+        <p>An Erlang <seeguide marker="system/reference_manual:data_types#list">list</seeguide>
+          containing terms of the type <c><anno>ContentType</anno></c> that may end with
+          a term of type <c><anno>TerminationType</anno></c>.
+        </p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="mfa" n_vars="0"/>
+      <desc>
+        <p>An three-tuple represening an Module:Function/Arity function signature.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="module" n_vars="0"/>
+      <desc>
+        <p>An Erlang module represented by an atom.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="neg_integer" n_vars="0"/>
+      <desc>
+        <p>A negative integer.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="nil" n_vars="0"/>
+      <desc>
+        <p>The empty <seetype marker="#list">list()</seetype>.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="no_return" n_vars="0"/>
+      <desc>
+        <p>The type used to show that a function will <em>never</em> return a value,
+          that is it will <em>always</em> throw an exception.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="node" n_vars="0"/>
+      <desc>
+        <p>An Erlang <seeguide marker="system/reference_manual:distributed#nodes">node</seeguide>
+          represented by an atom.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="non_neg_integer" n_vars="0"/>
+      <desc>
+        <p>A non-negative integer, that is any positive integer or 0.</p>
+      </desc>
+    </datatype>
+    <datatype>
+      <name name="none" n_vars="0"/>
+      <desc>
+        <p>The bottom type, contains no terms.</p>
```

Comment:
```suggestion
        <p>This type is used to show that a function will <em>never</em> return a value;
          that is it will <em>always</em> throw an exception. In a spec, use <c>no_return()</c>
          for the sake of clarity.</p>
```